src/Integration.UnitTests/LocalServices/ErrorListInfoBarController.QualityProfileBackgroundProcessorTests.cs

```diff
@@ -0,0 +1,386 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="ErrorListInfoBarController.QualityProfileBackgroundProcessorTests.cs" company="SonarSource SA and Microsoft Corporation">
```

Comment:
Separate test class for ErrorListInfoBarController.QualityProfileBackgroundProcessor.
I also added a test at the ErrorListInfoBarController level that checks for cancellation and have another that will also invoke the background processor and will no-op before executing the task.